python/uv/_build_backend.py

```diff
@@ -18,17 +18,17 @@
 
 TYPE_CHECKING = False
 if TYPE_CHECKING:
-    from typing import Any  # noqa:I001
+    from typing import Any, Mapping  # noqa:I001
 
 
-def warn_config_settings(config_settings: "dict[Any, Any] | None" = None) -> None:
+def warn_config_settings(config_settings: "Mapping[Any, Any] | None" = None) -> None:
     import sys
 
     if config_settings:
         print("Warning: Config settings are not supported", file=sys.stderr)
 
 
-def call(args: "list[str]", config_settings: "dict[Any, Any] | None" = None) -> str:
+def call(args: "list[str]", config_settings: "Mapping[Any, Any] | None" = None) -> str:
```

Comment:
`list` is invariant, so it `list[str]` won't accept `list[LiteralString]` or other lists whose value type is strict subtype of `str`.

Repro:
https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2AyjCKmgLABQLLAJicFMABQD6ALiiUkAZxgBtCSAC6AGigB6AJRQAtAD4oAOTAoSwgHQm2zSmOGiJkoqQo06DWVAC8USQCJPslnwsqZkA

So if this wasn't intended, then you could use `collections.abc.Sequence`, which has a covariant value type parameter.

Reply:
done in 7250bdbbd

Reply:
Is sequence also available in collections.abc since 3.8?